### PR TITLE
Fixed wrong application domain selected

### PR DESCRIPTION
**Pull request by DavidEshtehari**
In the process of starting the listener to receive the notifications from SQL Server instance, if there is not a process dispatcher, we should obtain it first. In this process, we need the main application domain in the active process that has access to `Microsoft.Data.SqlClient` assembly to create an instance of `SqlDependencyProcessDispatcher` type.
The issue occurred when we tried to create the instance on the wrong domain.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlDependencyTest/SqlDependencyTest.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlDependencyTest/SqlDependencyTest.cs]: Don't forget to include the license header at the top of any new files.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlDependencyTest/SqlDependencyTest.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests
+{
```
[the remark this answers is not in the document]
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlDependencyTest/SqlDependencyTest.cs]: This test can be universally applied, we only need to restrict testing if the other framework doesn't support something or needs a fix.

**Comment by DavidEshtehari** [comment]
In the process of starting the listener to receive the notifications from SQL Server instance, if there isn’t a process dispatcher, we should obtain it first. In this process we need the main application domain in the active process that has the access to Microsoft.Data.SqlClient assembly to create an instance of SqlDependencyProcessDispatcher type.
The issue occurred when we tried to create the instance on a wrong domain.

The complementary description enclosed in the attached file!
[SqlDependency-issue401-description.docx](https://github.com/dotnet/SqlClient/files/4173078/SqlDependency-issue401-description.docx)

**Comment by cheenamalhotra** [comment]
Uploading package for testing:
[Microsoft.Data.SqlClient.2.0.0-PR410.nupkg.zip](https://github.com/dotnet/SqlClient/files/4378011/Microsoft.Data.SqlClient.2.0.0-PR410.nupkg.zip)

**Pull request merged**